python/cudf/io/json.py

```diff
@@ -8,13 +10,49 @@
 
 
 @ioutils.doc_read_json()
-def read_json(path_or_buf, *args, **kwargs):
-    """{docstring}"""
+def read_json(path_or_buf, dtype=None, lines=False, compression='infer',
+              byte_range=None, *args, **kwargs):
 
-    warnings.warn("Using CPU via Pandas to read JSON dataset, this may "
-                  "be GPU accelerated in the future")
-    pd_value = pd.read_json(path_or_buf, *args, **kwargs)
-    return cudf.from_pandas(pd_value)
+    """
+    Load and parse a JSON file into a DataFrame
+
+    Parameters
+    ----------
+    path_or_buf : str
+        Path of file to be read or a str containing the file.
+    dtype : list of str or dict of {{col: dtype}}, default None
+        List of data types in the same order of the column names
+        or a dictionary with column_name:dtype (Pandas style).
+    lines : bool, default False
+        Read the file as a json object per line
+    compression : {{'infer', 'gzip', 'zip', None}}, default 'infer'
+        For on-the-fly decompression of on-disk data. If ‘infer’, then detect
+        compression from the following extensions: ‘.gz’,‘.zip’ (otherwise no
+        decompression). If using ‘zip’, the ZIP file must contain only one
+        data file to be read in, otherwise the first non-zero-sized file will
+        be used. Set to None for no decompression.
+    byte_range : list or tuple, default None
+        Byte range within the input file to be read. The first number is the
+        offset in bytes, the second number is the range size in bytes. Set the
+        size to zero to read all data after the offset location. Reads the row
+        that starts before or at the end of the range, even if it ends after
+        the end of the range.
+
+    Returns
+    -------
+    GPU ``DataFrame`` object.
+    """
+
+    if lines:
+        df = cpp_read_json(path_or_buf, dtype, lines, compression, byte_range)
```

Comment:
We should provide an option to still fall back to the Pandas JSON reader for JSONLines if the user wants. Could we add an `engine` keyword arg similar to the Parquet / ORC reader?

Reply:
Added 'engine' parameter with the same semantics as in read_orc and read_parquet. The default is still cuDF, when reading JSON Lines.

Reply:
I suggest either giving a info/warning if the user does not specify `lines` but uses `cudf`, or adding that info in the `docstring`.

Reply:
I can't add a warning because 'cudf' is the default engine, so I can't tell if the user specifically requested that engine or if they are just using the default version.
Added info to the docstring.

Reply:
It's more the fact that it's confusing when the default engine is `cudf` but that is only invoked when lines is `True` (it's default `False`).

So by default if a user tries to read a normal JSON file, the the engine is `cudf` but it's actually using PANDAS backend and it will print the warning CPU warning message...